bigquery/google/cloud/bigquery/dataset.py

```diff
@@ -486,30 +485,6 @@ def _build_resource(self):
 
         return resource
 
-    def exists(self, client=None):
```

Comment:
@jba Why don't you just make `client` required?

Reply:
The idea is to turn `Dataset` into more of a dumb object; all the RPCs will be on `Client`. Passing a client back into `Dataset` so it can do the call seems to be going in the other direction. Put another way, if we had `Dataset.exists(client)`, why wouldn't we have `Dataset.create(client)` and so on?

Reply:
SGTM

Reply:
Actually, I'm not sure it SGTM (sorry for the quick trigger).

This entire (15+ packages) `google.cloud` effort (over 3-ish years) has been pretty consistent in its approach. We have "nouns" for data and any "verbs" that operate on those nouns are methods on the corresponding class.

Reply:
Danny, could you review the "BigQuery Python Client Libraries Redesign Proposal"? I've made many similar proposals throughout the client library and would appreciate your feedback.

Reply:
The solution I proposed was something closer to what is being done for the newer auto-gen APIs with dumb resource objects and API calls being done from the client.

Reply:
Sure thing (don't hold up this PR on my account).